src/watchmaker/workers/salt.py

```diff
@@ -980,12 +1005,17 @@ def install(self):
         salt_running = False
         salt_enabled = False
         salt_svc = 'salt-minion'
+        self.salt_call = SaltWindows._get_salt_call()
```

Comment:
Yes, it is necessary because _get_salt_call() method returns the full path of the salt-call executable, and self.salt_call is used later in the method to execute salt states. If self.salt_call is not properly set, the salt states may fail to execute.

Reply:
Hmm, but this is set already in the `__init__()` method. So this should be set already by the time anything is calling the `install()` method. The only time this would not be the case, is if something were _directly_ invoking `SaltWindows.install()`, which is not supported. The class should first be instantiated, which will set the `salt_call` attribute correctly, and then that instance of the worker can call the `install()` method.

Reply:
ah understand, will remove it !